docs/docs/modules/model_io/prompts/prompt_templates/few_shot.mdx

```diff
@@ -0,0 +1,294 @@
+# Few Shot Prompt Templates
+
+Few shot prompting is a prompting technique which provides the Large Language Model (LLM) with a list of examples, and then asks the LLM to generate some text following the lead of the examples provided.
+
+An example of this is the following:
+
+Say you want your LLM to respond in a specific format. You can few shot prompt the LLM with a list of question answer pairs so it knows what format to respond in.
+
+```txt
+Respond to the users question in the with the following format:
+
+Question: What is your name?
+Answer: My name is John.
+
+Question: What is your age?
+Answer: I am 25 years old.
+
+Question: What is your favorite color?
+Answer:
+```
+
+Here we left the last `Answer:` undefined so the LLM can fill it in. The LLM will then generate the following:
+
+```txt
+Answer: I don't have a favorite color; I don't have preferences.
+```
+
+### Use Case
+
+In the following example we're few shotting the LLM to rephrase questions into more general queries.
+
+We provide two sets of examples with specific questions, and rephrased general questions. The `FewShotChatMessagePromptTemplate` will use our examples and when `.format` is called, we'll see those examples formatted into a string we can pass to the LLM.
+
+```typescript
+import {
+  ChatPromptTemplate,
+  FewShotChatMessagePromptTemplate,
+} from "langchain/prompts";
+```
+
+```typescript
+const examples = [
+  {
+    input: "Could the members of The Police perform lawful arrests?",
+    output: "what can the members of The Police do?",
+  },
+  {
+    input: "Jan Sindel's was born in what country?",
+    output: "what is Jan Sindel's personal history?",
+  },
+];
+const examplePrompt = ChatPromptTemplate.fromTemplate(`Human: {input}
+AI: {output}`);
+const fewShotPrompt = new FewShotChatMessagePromptTemplate({
+  examplePrompt,
+  examples,
+  inputVariables: [], // no input variables
+});
+```
+
+```typescript
+const formattedPrompt = await fewShotPrompt.format({});
+console.log(formattedPrompt);
+```
+
+```typescript
+[
+  HumanMessage {
+    lc_namespace: [ 'langchain', 'schema' ],
+    content: 'Human: Could the members of The Police perform lawful arrests?\n' +
```

Comment:
I thought the chat version of this returned human + AI message pairs?

Reply:
Only if you input human & ai. If you just give it human it'll only return human and viceversa